executor/memtable_reader.go

```diff
@@ -903,14 +899,18 @@ func (e *hotRegionsHistoryRetriver) retrieve(ctx context.Context, sctx sessionct
 		Store:       tikvStore,
 		RegionCache: tikvStore.GetRegionCache(),
 	}
+	tz := sctx.GetSessionVars().Location()
+	allSchemas := sctx.GetInfoSchema().(infoschema.InfoSchema).AllSchemas()
```

Comment:
Should we use `ok` to assert the `infoschema.InfoSchema` first to prevent potential panic?

Reply:
Use `GetTxnInfoSchema` func avoid this convert, #30574.
```go
	allSchemas := sessiontxn.GetTxnManager(sctx).GetTxnInfoSchema().AllSchemas()
```

Reply:
```go
	// Deprecated: Use TxnManager.GetTxnInfoSchema to get the current schema in session
	GetInfoSchema() InfoschemaMetaVersion
```

Reply:
cc @JmPotato